microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/model/PropertyValueExpression.java

```diff
@@ -105,38 +118,97 @@ public String getReferencedPropertyName() {
 	 *         otherwise.
 	 */
 	public boolean isClosed() {
-		String text = getText();
-		return text.charAt(text.length() - 1) == '}';
+		int end = super.getEnd();
+		if (end == -1) {
+			return false;
+		}
+		return super.getOwnerModel().getText().charAt(end - 1) == '}';
 	}
 
 	/**
-	 * Returns the offset of the start of the referenced property name,
-	 * or the offset after the `$` if no property is referenced.
+	 * Returns the offset of the start of the referenced property name, or the
+	 * offset after the `$` if no property is referenced.
 	 * 
-	 * @return the offset of the start of the referenced property name,
-	 * or the offset after the `$` if no property is referenced.
+	 * @return the offset of the start of the referenced property name, or the
+	 *         offset after the `$` if no property is referenced.
 	 */
 	public int getReferenceStartOffset() {
-		String propName = getReferencedPropertyName();
-		if (propName == null) {
-			return getStart() + 1;
-		}
-		return getStart() + getText().indexOf(propName);
+		parseExpressionIfNeeded();
+		return referenceNameStartOffset;
 	}
 
 	/**
-	 * Returns the offset of the end of the referenced property name,
-	 * or the offset after the `$` if no property is referenced.
+	 * Returns the offset of the end of the referenced property name, or the offset
+	 * after the `$` if no property is referenced.
 	 * 
-	 * @return the offset of the end of the referenced property name,
-	 * or the offset after the `$` if no property is referenced.
+	 * @return the offset of the end of the referenced property name, or the offset
+	 *         after the `$` if no property is referenced.
 	 */
 	public int getReferenceEndOffset() {
-		String propName = getReferencedPropertyName();
-		if (propName == null) {
-			return getStart() + 1;
+		parseExpressionIfNeeded();
+		return referenceNameEndOffset;
+	}
+
+	public String getDefaultValue() {
+		parseExpressionIfNeeded();
+		if (hasDefaultValue()) {
+			return super.getOwnerModel().getText(defaultValueStartOffset, defaultValueEndOffset, true);
 		}
-		return getStart() + getText().indexOf(propName) + propName.length();
+		return null;
+	}
+
+	public boolean hasDefaultValue() {
+		return defaultValueStartOffset != -1 && defaultValueEndOffset != -1 && defaultValueStartOffset != defaultValueEndOffset;
 	}
 
+	private void parseExpressionIfNeeded() {
+		if (parsed) {
+			return;
+		}
+		parseExpression();
+	}
+
+	private synchronized void parseExpression() {
+		if (parsed) {
+			return;
+		}
+
+		int start = super.getStart();
+		int end = super.getEnd();
+		if (start == -1 || end == -1) {
+			return;
+		}
+		if (start >= end) {
+			return;
+		}
+		boolean nameParsing = true;
+		referenceNameStartOffset = start + 2;
+		String text = super.getOwnerModel().getText();
+		for (int i = referenceNameStartOffset; i < end; i++) {
+			char c = text.charAt(i);
+			switch (c) {
```

Comment:
I believe the offset ensures we start scanning after `${`, then the cases are treated accordingly. If we have a name as follows:
```
quarkus.application.name=${abc:def:XXX}
```
The hover will only recognize after the last :
![Screenshot from 2021-10-13 13-57-15](https://user-images.githubusercontent.com/26389510/137187875-1e840570-795a-455c-8c7a-a6d4c1b10386.png)

Should we permit this behaviour?